src/DynamicData/Kernel/InternalEx.cs

```diff
@@ -14,6 +14,12 @@ namespace DynamicData.Kernel;
 /// </summary>
 public static class InternalEx
 {
+#if NET9_0_OR_GREATER
```

Comment:
@JakenVeina / @dwcullop 

We may be able to use a poly fill, and use Lock everywhere.

Reply:
A "polyfill" as we've done before would require pulling the entire source code of .NET's `Lock` class into our own, and `#if` checking around that. Same for all the APIs that USE `Lock` that we'd want to consume.

The "polyfill"s that we have today work really well because those classes don't HAVE any actual code in them, nor any APIs that use them. The only thing that uses them is the compiler, for the sake of static analysis. on newer language features.

What we'd have to do to eliminate the `#if`s here would be to have an entire wrapper class of our own, that is implemented separately .NET 8- and .NET 9+. Which probably would end up beneficial. Would you want to take this PR first, and then circle back to that?

Reply:
It's probably not worth porting the Lock code. What do you think?